scripts/jenkins/cloud/ansible/start-deployer-vm.yml

```diff
@@ -18,7 +18,7 @@
 - name: Start fresh deployer VM
   hosts: "gate-{{ cloud_env }}"
   remote_user: root
-  gather_facts: no
+  gather_facts: yes
```

Comment:
I am not really sure about this change. Where do you need a fact from the gate node?

We should not need any information about the gate node. If you are using the `ansible_distribution_version` from it to get the `sles_cloud_version` it will occasionally fail because there is no relationship between them. There are gate nodes with SLES12SP3 and SLES12SP4 and both hosting the deployer nodes for cloud 8 and 9.

Reply:
So this was a two pronged fix - one to try and collect the facts so that ansible_distribution_version has a value, and another part to define an obviously erroreous value for it so that it will be easier to debug a failure in this area...

Another option would be to ensure that cloudsource has a meaningful value when ever it is passed in...

However the good news is that in the case of this particular playbook, start-deployer-vm.yml it appears that it will only be invoked with cloudsource that is meaningful, so if we prefer not to retain the gather_facts: yes setting, at least the falling back to using the default value of UNKOWN if ansible_distribution_version is not defined.

Reply:
I have updated the patch to remove the gather_facts: yes part.